test/server_test.cpp

```diff
@@ -0,0 +1,35 @@
+#include <sstream>
+#include <string>
+#include <iostream>
+
+#include "gtest/gtest.h"
+#include "server.hpp"
+#include <signal.h>
+
+
+TEST(ServerTest, PortTooBig) {
+	std::string addr = "0.0.0.0";
+	std::string port = "12312312";
+
+	http::server::server server_(addr, port);
+	EXPECT_FALSE(server_.isValid(addr, port));
+}
+
+TEST(ServerTest, PortIsNegative) {
```

Comment:
Yeah, the problem is when the server tries to construct itself with a negative port, an exception is thrown. isValid() never actually gets a chance to be called here.